src/fragments/lib/auth/js/mfa.mdx

```diff
@@ -14,81 +14,184 @@ With TOTP (Time-based One-time Password), your app user is challenged to complet
 
 You can setup TOTP for a user in your app:
 
-```javascript
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { CognitoUserSession } from 'amazon-cognito-identity-js';
 import { Auth } from 'aws-amplify';
 
-// To setup TOTP, first you need to get a `authorization code` from Amazon Cognito
-// `user` is the current Authenticated user
-Auth.setupTOTP(user).then((code) => {
+type SetupTOTPAuthParameters = {
+  user: string;
+  challengeAnswer: string;
+  mfaType?: 'SMS_MFA' | 'SOFTWARE_TOKEN_MFA' | null;
+};
+
+export async function setupTOTPAuth({ user, challengeAnswer, mfaType }: SetupTOTPAuthParameters) {
+  // To setup TOTP, first you need to get a `authorization code` from Amazon Cognito
+  // `user` is the current Authenticated user
+  const tOTPCode = await Auth.setupTOTP(user);
```

Comment:
Technically, `Auth.setupTOTP(user)` doesn't return the OTP code yet. It returns a secretCode that is used on authenticator Apps to generate the OTP code.